test/parallel/test-util-inspect.js

```diff
@@ -906,34 +906,30 @@ if (typeof Symbol !== 'undefined') {
   assert.strictEqual(util.inspect(resolved),
                      common.engineSpecificMessage({
                        v8: 'Promise { 3 }',
-                       chakracore: 'Promise {}'
```

Comment:
Why are we pending when V8 is resolved here (and rejected below?)

Reply:
is this because the Promise here isn't a natively-constructed `v8::Promise`, but a "bare" promise coming from script?

Reply:
Yup, that's exactly right.

Reply:
I wonder if we should change our `v8::Promise` implementation to detect not-a-v8-promise promises and give some kind of sign? As it stands, we really shouldn't be passing these couple of tests because the functionality doesn't exist.

Reply:
The good news is that none of these tests even run in master, I was fixing them up to at least verify the current behavior in v8.x and just figured that backporting would be a good idea.

@MSLaguana what would you suggest instead?

Reply:
I'm afraid I don't have a great suggestion for the immediate term. I was thinking that if we wanted to support this, we could add a new JSRT api to get the state of a promise, along with the value, and use that here (and in f12?).

Reply:
I'll just split this change out for now.

Reply:
@MSLaguana I moved this over to #474.  Let's continue over there.